core/parameters.class.inc.php

```diff
@@ -55,13 +55,14 @@ protected function ToXML(DOMNode $oRoot, $data = null, $sNodeName = null)
 			$data = $this->aData;
 		}
 
+		if ($oRoot instanceof DOMDocument) {
+			$oNode = $oRoot->createElement($sNodeName);
+		} else {
+			$oNode = $oRoot->ownerDocument->createElement($sNodeName);
+		}
+		$oRoot->appendChild($oNode);
+
```

Comment:
Why was this block moved outside of the `if (is_array($data))` test ?

Reply:
Because it is also needed for the `else`